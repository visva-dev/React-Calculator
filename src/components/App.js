import React from 'react';
import '../app.css';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import Calculate from '../logic/Calculate';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
      operation: null,
    };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(buttonName) {
    const { total, next, operation } = this.state;
    const data = { total, next, operation };

    const result = Calculate(data, buttonName);

    this.setState({
      total: result.total,
      next: result.next,
      operation: result.operation,
    });
  }

  render() {
    const { total, next, operation } = this.state;
    let result;
    if (operation === null) {
      result = total;
    } else if (operation === '+/-') {
      if (next === null || next === '0') {
        result = total;
      } else {
        result = next;
      }
    } else {
      result = next === null ? operation : next;
    }

    return (
      <div className="app">
        <Display result={result} />
        <ButtonPanel clickHandler={this.handleClick} />
      </div>
    );
  }
}

export default App;
