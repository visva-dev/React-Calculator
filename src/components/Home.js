import React from 'react';
import NavBar from './Navbar';

const Home = () => (
  <>
    <NavBar />
    <h1>Math Magicians</h1>
    <p>
      Lorem ipsum dolor, sit amet consectetur adipisicing elit. Consequuntur,
      quis delectus. Aliquid qui recusandae, maiores tempora perspiciatis
      officiis nobis reprehenderit blanditiis dolor, iure ad odit rem iusto,
      natus necessitatibus perferendis unde veritatis consequatur assumenda quo
      cumque atque ratione alias? Vero molestias facere animi dolores pariatur
      debitis quos at consectetur laudantium, similique, itaque, nesciunt
      veritatis ex?
    </p>
  </>
);

export default Home;
