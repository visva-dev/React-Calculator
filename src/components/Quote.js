import React from 'react';
import NavBar from './Navbar';

const Quote = () => (
  <>
    <NavBar />
    <h1>Every programmer is an author.</h1>
  </>
);

export default Quote;
