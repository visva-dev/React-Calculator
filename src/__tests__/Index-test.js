import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import renderer from 'react-test-renderer';
import App from '../components/App';
import Home from '../components/Home';
import Quote from '../components/Quote';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Switch>
          <Route path="/" exact component={Home} />
          <Route path="/App" exact component={App} />
          <Route path="/Quote" exact component={Quote} />
        </Switch>
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
