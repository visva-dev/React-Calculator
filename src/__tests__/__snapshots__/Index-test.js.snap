// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="navbar"
  >
    <h1>
      Math Magicians
    </h1>
    <div
      className="nav"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home 
      </a>
      <a
        href="/App"
        onClick={[Function]}
      >
        Calculator 
      </a>
      <a
        href="/Quote"
        onClick={[Function]}
      >
        Quote 
      </a>
    </div>
  </div>,
  <h1>
    Welcome to our page!
  </h1>,
  <p>
    Lorem ipsum dolor, sit amet consectetur adipisicing elit. Consequuntur, quis delectus. Aliquid qui recusandae, maiores tempora perspiciatis officiis nobis reprehenderit blanditiis dolor, iure ad odit rem iusto, natus necessitatibus perferendis unde veritatis consequatur assumenda quo cumque atque ratione alias? Vero molestias facere animi dolores pariatur debitis quos at consectetur laudantium, similique, itaque, nesciunt veritatis ex?
  </p>,
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Tenetur illum dicta libero voluptatum vitae molestias, porro corrupti sit corporis omnis architecto error quod fuga temporibus assumenda! Saepe, voluptatum id neque a veritatis eveniet quos molestiae, doloribus consectetur eaque autem fugiat ex architecto accusantium. Libero odio impedit deserunt quasi expedita? Quidem reiciendis illum porro minus voluptates ut, quibusdam cumque ducimus suscipit!
  </p>,
]
`;
